packages/dev/test/RelayServer.test.ts

```diff
@@ -605,6 +606,212 @@ contract('RelayServer', function (accounts: Truffle.Accounts) {
+      configFilename = `${env.relayServer.config.workdir}/withdraw.json`
+      assert.isFalse(fs.existsSync(configFilename))
+      currentBlockNumber = await env.web3.eth.getBlockNumber()
+    })
+    afterEach(async function () {
+      sinon.restore()
+    })
+    it('should not withdraw if relayer is not ready', async function () {
+      fs.writeFileSync(configFilename, onceWithdrawConfig)
+      env.relayServer.setReadyState(false)
+      const serverSpy = sinon.spy(env.relayServer)
+      const txHashes = await env.relayServer.withdrawToOwnerIfNeeded(currentBlockNumber)
+      assert.deepEqual(txHashes, [])
+      sinon.assert.calledOnce(serverSpy.isReady)
+      assert.isFalse(serverSpy.isReady.returnValues[0])
+    })
+
+    it('should not withdraw if missing config file', async function () {
+      const serverSpy = sinon.spy(env.relayServer)
+      const txHashes = await env.relayServer.withdrawToOwnerIfNeeded(currentBlockNumber)
+      assert.deepEqual(txHashes, [])
+      sinon.assert.calledOnce(serverSpy.isReady)
+      assert.isTrue(serverSpy.isReady.returnValues[0])
+    })
+
+    describe('_resolveWithdrawalConfig', function () {
+      async function assertReturnedError (message: string): Promise<void> {
+        const spy = sinon.spy(env.relayServer.logger, 'error')
+        await env.relayServer.withdrawToOwnerIfNeeded(currentBlockNumber)
+        sinon.assert.calledWith(spy, `withdrawToOwnerIfNeeded: ${message}`)
+        spy.restore()
+      }
+      it('should throw with bad config file', async function () {
+        const badWithdrawConfig = JSON.stringify({
+          withdrawOnEthBalanceReached: 1e18,
+          leaveManagerWithAmountEth: 0.5e18,
+          repeat: false,
+          blah: 'blah'
+        })
+        fs.writeFileSync(configFilename, badWithdrawConfig)
+        try {
+          await env.relayServer._resolveWithdrawalConfig(configFilename)
+          assert.fail()
+        } catch (e) {
+          assert.include(e.message, 'Did not expect property `blah` to exist, got `blah` in object `withdrawConfig`')
+          await assertReturnedError(e.message)
+        }
+      })
+
+      it('should throw if withdrawOnEthBalanceReached < leaveManagerWithAmountEth', async function () {
+        const badWithdrawConfig = JSON.stringify({
+          withdrawOnEthBalanceReached: 0.5e18,
+          leaveManagerWithAmountEth: 1e18,
+          repeat: false
+        })
+        fs.writeFileSync(configFilename, badWithdrawConfig)
+        try {
+          await env.relayServer._resolveWithdrawalConfig(configFilename)
+          assert.fail()
+        } catch (e) {
+          assert.include(e.message, 'withdrawOnEthBalanceReached must be at least leaveManagerWithAmountEth')
+          await assertReturnedError(e.message)
+        }
+      })
+
+      it('should throw if leaveManagerWithAmountEth < managerTargetBalance', async function () {
+        const badWithdrawConfig = JSON.stringify({
```

Comment:
`managerTargetBalance` is the initial server config param, which is the amount of eth the server should maintain when withdrawing from the hub to itself. `leaveManagerWithAmountEth` is the amount you leave the manager after withdrawing to the owner. I thought that it might be larger, since you might want to leave a bit more to the server when withdrawing once to the owner. I can remove it if unnecessary.

Reply:
Not relevant anymore since it was removed.